mifosng-provider/src/integrationTest/java/org/mifosplatform/integrationtests/SearchTest.java

```diff
@@ -0,0 +1,230 @@
+					    	 System.out.println("------------------------EXACT SEARCH CENTER BY"+"  "+"ExternalId"+"="+externalId+"------------------------------\n");
+					    	 System.out.println(exactMatchCenterDetail_Search_By_ExternalId+"\n");
+						     
+						     ArrayList<HashMap> exactMatchCenterDetail_Search_By_Name =SearchHelper.exactMatchCenterSearch(requestSpec, responseSpec, displayName);
+						     System.out.println("------------------------EXACT SEARCH CENTER BY"+"  "+"groupName"+"="+displayName+"------------------------------\n");
+						     System.out.println(exactMatchCenterDetail_Search_By_Name+"\n");
+						     
+						     ArrayList<HashMap> LikeMatchCenterDetail_Search_By_CenterId=SearchHelper. likeMatchCenterSearch(requestSpec, responseSpec, likeMatch_displayName);
+				    	     System.out.println("-------------------------LIKE SEARCH CENTERS BY"+"  "+"likeMatch_displayName"+"="+likeMatch_displayName+"------------------------------\n");    	
+						     System.out.println(LikeMatchCenterDetail_Search_By_CenterId+"\n");
+						     
+						     ArrayList<HashMap> LikeMatchCenterDetail_Search_By_CenterName=SearchHelper. likeMatchCenterSearch(requestSpec, responseSpec,likeMatch_groupId );
+				    	     System.out.println("-------------------------LIKE SEARCH CENTERS BY"+"  "+"likeMatch_externalId"+"="+likeMatch_externalId+"------------------------------\n");    	
+						     System.out.println(LikeMatchCenterDetail_Search_By_CenterName+"\n");		     
+					    	
+					    	 ArrayList<HashMap> LikeMatchCenterDetail_Search_By_ExternalId=SearchHelper. likeMatchCenterSearch(requestSpec, responseSpec,likeMatch_externalId );
+				    	     System.out.println("-------------------------LIKE SEARCH CENTERS BY"+"  "+"likeMatch_externalId"+"="+likeMatch_externalId+"------------------------------\n");    	
+					    	 System.out.println(LikeMatchCenterDetail_Search_By_ExternalId+"\n");
+				     }
+		/***********************************************************Global Search****************************************************************/
+				     
+				     String search_parameter="000000001";
+				     String like_search_parameter=search_parameter.substring(1,3);
+				     
+				     ArrayList<HashMap> exactMatch_Global_Search_detail=SearchHelper.exactMatchGlobalSearch(requestSpec, responseSpec, search_parameter);
+		    	     System.out.println("-------------------------LIKE GLOBAL SEARCH BY"+"  "+"search_parameter"+"="+search_parameter+"------------------------------");    	
+				     System.out.println(exactMatch_Global_Search_detail);   
+
+				     ArrayList<HashMap> LikeMatch_Global_Search_detail=SearchHelper.likeMatchGlobalSearch(requestSpec, responseSpec, like_search_parameter);
+		    	     System.out.println("-------------------------LIKE GLOBAL SEARCH BY"+"  "+"like_search_parameter"+"="+like_search_parameter+"------------------------------");    	
```

Comment:
if groupName = "Group1"
Exact search should be validated as below
assertEquals(groupName, resultMap.get("groupName")
In case of  exact search = false , groupName = "Group"
We need to validate if resultMap contains groupNames containing substring "Group".We may get more than one groupname in this case and hence need to do this substring check for all groupnames fetched, to ensure we are not getting any groupname without substring "Group"